docs/markdown/Syntax.md

```diff
@@ -588,3 +588,64 @@ FAQ](FAQ.md#why-is-meson-not-just-a-python-module-so-i-could-code-my-build-setup
 because of this limitation you find yourself copying and pasting code
 a lot you may be able to use a [`foreach` loop
 instead](#foreach-statements).
+
+Grammar
+--
+
+This is the full Meson grammar, as it is used to parse Meson build definition files:
+
+```
+program = (NEWLINE | stmt)*
+stmt = (expression_stmt | selection_stmt | iteration_stmt | jump_stmt) NEWLINE
+selection_stmt = 'if' condition NEWLINE (stmt)* ('elif' condition NEWLINE (stmt)*)* ['else' (stmt)*] 'endif'
+iteration_stmt = 'foreach' identifier_list ':' identifier NEWLINE (stmt)* 'endforeach'
+condition = conditional_expr // TODO classic way would be condition = expression, but we can't have assignments in conditions. Should that be covered by the grammar (syntax) or not
+identifier_list = identifier | (identifier_list ',' identifier)
+jump_stmt = 'break' | 'continue' // TODO can currently only occur inside a iteration_stmt body, but that is not reflected here
+parameter_list = [arg_list] | [kwarg_list] | (arg_list ',' kwarg_list) // TODO could be more concise
+arg_list = expr_list
+kwarg_list = kwarg | (kwarg_list ',' kwarg)
+kwarg = string_literal ':' expr // TODO or could the key here actually be an expression?
```

Comment:
The answer is yes this can be an expression

Reply:
The question "what kind of expression is allowed here" appears in multiple places. I think I go with the base expression, which can be assignment expression. Something lilke `func(kwarg: a = 'abc')` is syntactically valid, it's just that Meson evaluates assignment expressisions to `None`, so the type system errors out here.